### PR TITLE
secure flag cookies

### PR DESCRIPTION
**Pull request by sidgogyan**
### What do you want to achieve?
-add secure flag cookies

### PR Checklist

- [x] I have tested it locally and all functionalities are working fine.
- [x] I have compared it with mocks and all design elements are the same.
- [ ] I have tested it in IE Browser.

### How was the code tested?

-tested on locally

### What new thing you came across while writing this code? 
-

### In case you fixed a bug then please describe the root cause of it? 
-

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: webplugin/js/app/km-utils.js [hunk 1]
```diff
@@ -257,10 +257,7 @@ KommunicateUtils = {
                 : cookiePrefix + cookie.name;
         var value = cookie.value;
         var path = '/';
-        var secure =
-            typeof cookie.secure == 'undefined'
-                ? this.isHttpsEnabledConnection()
-                : cookie.secure;
+        var secure =true;
```
Comment (devashishmamgain) [inline webplugin/js/app/km-utils.js:260]: is there any issue with the customer's site itself is in HTTP, will everything work fine for them?

File: webplugin/js/app/km-utils.js [hunk 2]
```diff
@@ -286,7 +283,7 @@ KommunicateUtils = {
             path +
             (secure ? ';secure' : '') +
             (domain ? ';domain=' + domain : '') +
-            (isChrome ? ';samesite=lax' : '');
+            (';SameSite=Strict');
```
Comment (Prochnost55) [inline webplugin/js/app/km-utils.js:286]: https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie
Screenshot 2022-07-07 at 12 42 49 PM

should it be SameSite or samesite?

**Review by Prochnost55 — CHANGES_REQUESTED** [review]

File: webplugin/js/app/km-utils.js (hunk 2 above)
Comment (sidgogyan) [reply webplugin/js/app/km-utils.js]: done

File: webplugin/js/app/km-utils.js (hunk 1 above)
Comment (sidgogyan) [reply webplugin/js/app/km-utils.js]: i have changed code to their previous state

**Review by Prochnost55 — APPROVED** [review]

File: webplugin/js/app/km-utils.js (hunk 1 above)
Comment (sidgogyan) [reply webplugin/js/app/km-utils.js]: this will add attribute secure over HTTPS and this samsite attribute of cookies will be strict

**Comment by devashishmamgain** [comment]
@sidgogyan @Prochnost55 
could you please add details on why strict and not None?

please check this
Screenshot 2022-07-08 at 10 56 49 AM


https://web.dev/samesite-cookies-explained/

**Comment by sidgogyan** [comment]
> @sidgogyan @Prochnost55 could you please add details on why strict and not None?
> 
> please check this Screenshot 2022-07-08 at 10 56 49 AM
> 
> https://web.dev/samesite-cookies-explained/

Sir, as per the MDN docs

[Strict](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite#strict)
Cookies will only be sent in a first-party context and not be sent along with requests initiated by third party websites.

[None](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite#none)
Cookies will be sent in all contexts, i.e. in responses to both first-party and cross-site requests. If SameSite=None is set, the cookie [Secure](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#secure) attribute must also be set (or the cookie will be blocked).

I thought it is not a good idea to share these cookies with third parties as these cookies are used at kommunicate side only. Therefore using strict.
Screenshot 2022-07-08 at 11 19 25 AM

**Review by devashishmamgain — APPROVED** [review]

**Pull request merged**